<?xml version="1.0"?>
<launch>
  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <arg name="input_pointcloud" default="/perception/obstacle_segmentation/pointcloud"/>
  <arg name="input_planning_trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="output_filtered_objects" default="/perception/object_recognition/filtered_objects"/>
  <arg name="output_filtered_pointcloud" default="/perception/obstacle_segmentation/filtered_pointcloud"/>
  <arg name="debug_filtering_markers" default="/perception_filter_node/debug/filtering_markers"/>
  <arg name="debug_object_markers" default="/perception_filter_node/debug/object_markers"/>
  <arg name="debug_objects_processing_time" default="/perception_filter_node/debug/objects_processing_time_ms"/>
  <arg name="debug_pointcloud_processing_time" default="/perception_filter_node/debug/pointcloud_processing_time_ms"/>
  <arg name="debug_processing_time_detail" default="/perception_filter_node/debug/processing_time_detail"/>
  <arg name="perception_filter_param_path" default="$(find-pkg-share autoware_launch)/config/planning/perception_filter/perception_filter.param.yaml"/>

  <node pkg="autoware_perception_filter" exec="autoware_perception_filter_node" name="perception_filter_node" output="both">
    <remap from="input/objects" to="$(var input_objects)"/>
    <remap from="input/pointcloud" to="$(var input_pointcloud)"/>
    <remap from="input/planning_trajectory" to="$(var input_planning_trajectory)"/>
    <remap from="output/filtered_objects" to="$(var output_filtered_objects)"/>
    <remap from="output/filtered_pointcloud" to="$(var output_filtered_pointcloud)"/>
    <remap from="debug/filtering_markers" to="$(var debug_filtering_markers)"/>
    <remap from="debug/object_markers" to="$(var debug_object_markers)"/>
    <remap from="debug/objects_processing_time_ms" to="$(var debug_objects_processing_time)"/>
    <remap from="debug/pointcloud_processing_time_ms" to="$(var debug_pointcloud_processing_time)"/>
    <remap from="debug/processing_time_detail" to="$(var debug_processing_time_detail)"/>
    <param from="$(var perception_filter_param_path)"/>
  </node>
</launch>
